pkg/controller/services.go

```diff
@@ -684,6 +711,16 @@ func (cont *AciController) updateServiceDeviceInstance(key string,
 	}
 
 	graphName := cont.aciNameForKey("svc", "global")
+	deviceName := cont.aciNameForKey("svc", "global")
```

Comment:
How would you pass a different name here? With another annotation?

Reply:
If user annotates the service, we will check the tnVnsAbsGraphName in the contract (https://github.com/noironetworks/aci-containers/pull/1075/files#diff-d5c31f7209190065692316305836fc8a42834003000db2714b32ac867cc5c96fR717) and use the same graphName that is there in apic

Reply:
This question was why we need two arguments into ApicDevCtx, if we only use devicename argument.

Reply:
apicDevCtx has reference of both deviceName and graphName. 
https://github.com/noironetworks/aci-containers/pull/1075/files#diff-d5c31f7209190065692316305836fc8a42834003000db2714b32ac867cc5c96fR576 - Here we use graphName
https://github.com/noironetworks/aci-containers/pull/1075/files#diff-d5c31f7209190065692316305836fc8a42834003000db2714b32ac867cc5c96fR578 - Here we use deviceName 

When we create a global SG template and device, we create it with same name <aci_prefix>_global_svc. So, we were passing just graphName (since graphName was same as deviceName) to the function apicDevCtx. But, in case of custom SG, device remains same(<aci_prefix>_global_svc) and graphName will be different. So, we had to pass both deviceName and graphName to apicDevCtx